### PR TITLE
Add travis test

### LINKED ISSUE #17: make this Erlang 21.X compatible

### PR DESCRIPTION
**Pull request by getong**
fix #17 and #22

### CODE REVIEW

**Comment by getong** [comment]
The common tests build failed, I have no aws environment.

**Comment by velimir** [comment]
tests do not request AWS environment

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sudo: false
+language: erlang
+
+otp_release:
+    - 21.1.3
+    - 20.3.8
+
+install:
+   - wget -c https://github.com/erlang/rebar3/releases/download/3.7.5/rebar3 && chmod +x rebar3
```
Comment (motobob) [inline .travis.yml]: i'll need to re-verify anything above 3.6.2... 3.7.X broke many plugins and i want to make sure our `rebar3_erllamba` will not

File: src/erllambda_poller.erl [hunk 2]
```diff
@@ -115,7 +115,7 @@ handle_info(poll, #state{handler = Handler} = State) ->
     {ok, ReqId, Hdrs, Body} = invoke_next(State),
     %% container thaw happens here
     erllambda:message("Next returns, in invoke ~p", [os:system_time(millisecond)]),
-    case erllambda:invoke(Handler, Body, Hdrs) of
+    case erllambda:invoke(Handler, Body, maps:to_list(Hdrs)) of
```
Comment (motobob) [inline src/erllambda_poller.erl]: `Hdrs` is list here and all map conversions deliberately to happen in handle process so not to pile up small binaries in poller between invokes

File: src/erllambda.erl [hunk 3]
```diff
@@ -325,11 +326,11 @@ invoke_exec( Handler, Event, Context ) ->
             {unhandled, Response}
     end.
 
-invoke_credentials( #{<<"AWS_SESSION_TOKEN">> := Token} = _Context ,
-                    {ok, #aws_config{security_token = Token}}) ->
+invoke_credentials( #{<<"AWS_SESSION_TOKEN">> := Token1} = _Context ,
```
Comment (motobob) [inline src/erllambda.erl]: what is the purpose of this change?

Comment (getong) [reply src/erllambda.erl]: In the code
```
Context = maps:merge(os_env2map(), hdr2map(EventHdrs)),
os_env2map() ->
    maps:from_list(lists:map(
        fun(S) ->
            [K, V] = string:split(S, "="),
            {list_to_binary(K), list_to_binary(V)}
        end,
        os:getenv()
    )).
hdr2map(Hdrs) ->
    maps:from_list([{list_to_binary(K), list_to_binary(V)} || {K, V} <- Hdrs]).
```
The `Context` variable is a map, which key type is binary, and its value is also binary.
in the `erlcloud/include/erlcloud_aws.hrl` head file:
```
security_token=undefined::string()|undefined,
```
So, `Token1` is binary, and `Token2` is `string()` or `undefined`, it never euqals.
Just changing the type of writing to ignore the dialyzer error.

File: .travis.yml (hunk 1 above)
Comment (getong) [reply .travis.yml]: I think it is the same, but if needed, I can change it to 3.6.2.

**Comment by getong** [comment]
In the build test log, https://travis-ci.org/getong/erllambda/builds/462211570 
it tests ok.

File: src/erllambda.erl (hunk 3 above)
Comment (velimir) [reply src/erllambda.erl]: this looks like a bug, and should be fixed, rather than hidden.

**Comment by motobob** [comment]
my thinking is we just take it in as is but in future ask to do separate PRs

**Comment by motobob** [comment]
@getong if you have a chance to rebase it  please do so, if not we will take your changes to our branch and merge it all.
this is great thing to have in.

**Comment by motobob** [comment]
done under #49 right @velimir ?

**Comment by velimir** [comment]
@motobob, correct. Closing. @getong thank you for the contribution! Please feel free to reopen pr if you think we missed something.
